tools/Docker-dev/Dockerfile

```diff
@@ -0,0 +1,143 @@
+# 
+# Dockerfile for Scipy development 
+# 
+# Usage: 
+# -------
+# 
+# To make a local build of the container, from the 'Docker-dev' directory:
+# docker build  --rm -f "Dockerfile" -t <build-tag> "."    
+# 
+# To use the container use the following command. It assumes that you are in
+# the root folder of the scipy git repository, making it available as
+# /home/scipy in the container. Whatever changes you make to that directory
+# are visible in the host and container.
+# The docker image is retrieved from the scipy dockerhub repository
+#
+# docker run --rm -it -v $(pwd):/home/scipy scipy:<image-tag>
+# 
+# Once the container is running, you'll need to activate the conda environment
+# conda activate scipydev
+# 
+# And run the tests with
+# python runtests.py
+# 
+# Ubuntu 20.04 (focal)
+# https://hub.docker.com/_/ubuntu/?tab=tags&name=focal
+# OS/ARCH: linux/amd64
+ARG ROOT_CONTAINER=ubuntu:focal-20201106@sha256:4e4bc990609ed865e07afc8427c30ffdddca5153fd4e82c20d8f0783a291e241
+ARG BASE_CONTAINER=${ROOT_CONTAINER}
+
+# hadolint ignore=DL3006 as base container is tagged
+FROM ${BASE_CONTAINER}
+
+# Change default shell - this avoids issues with Conda later
+SHELL ["/bin/bash","--login", "-o", "pipefail", "-c"]
+
+# -----------------------------------------------------------------------------
+# ---- Miniforge installer ----
```

Comment:
Just a note: I checked the time the solve takes, since that can be problematic in CI. Here it's ~12 seconds, which is fine.